lib/client/extended_keyring.go

```diff
@@ -0,0 +1,528 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was partially copied from x/crypto/ssh/agent.keyring.go
```

Comment:
Are they any chances that instead of copying the file, we could just extend the existing implementation on the Go library, but for example, embedding the object?

Reply:
This is primarily an extension of the keyring in `x/crypto/ssh/agent.keyring.go`, the only thing copied is the locking mechanism, the error messages, and a few other small things. I'm not sure whether the mention to `x/crypto/ssh/agent.keyring.go` is necessary at all, let me know what you think.